src/core/solvers/ipopt/ipopt-iface.cpp

```diff
@@ -0,0 +1,665 @@
+          // We need the lower triangular matrix
+          if (idx_col > idx_row) {
+            break;
+          }
+          iRow[idx_value] = ixu + idx_row;
+          jCol[idx_value] = ixu + idx_col;
+          idx_value++;
+        }
+      }
+    }
+
+    assert_pretty(idx_value == nele_hess,
+                  "Number of Hessian elements set does not coincide with the total non-zero Hessian values");
+  } else {
+    // return the values. This is a symmetric matrix, fill the lower left
+    // triangle only
+    std::size_t idx_value = 0;
+    std::size_t ixu = 0;
+    // Running Costs
+    for (std::size_t t = 0; t < T_; t++) {
+      const boost::shared_ptr<ActionModelAbstract> &model = problem_->get_runningModels()[t];
+      const boost::shared_ptr<ActionDataAbstract> &data = problem_->get_runningDatas()[t];
+      const std::size_t ndxi = model->get_state()->get_ndx();
+      const std::size_t nui = model->get_nu();
+
+      datas_[t]->dx = Eigen::VectorXd::Map(x + ixu, ndxi);
+      datas_[t]->u = Eigen::VectorXd::Map(x + ixu + ndxi, nui);
+
+      model->get_state()->integrate(xs_[t], datas_[t]->dx, datas_[t]->x);
+
+      model->calcDiff(data, datas_[t]->x, datas_[t]->u);  // this might be removed
+
+      model->get_state()->Jintegrate(xs_[t], datas_[t]->dx, datas_[t]->Jsum_dx, datas_[t]->Jsum_dx, second, setto);
+      datas_[t]->Ldxdx = datas_[t]->Jsum_dx.transpose() * problem_->get_runningDatas()[t]->Lxx * datas_[t]->Jsum_dx;
+      datas_[t]->Ldxu = datas_[t]->Jsum_dx.transpose() * problem_->get_runningDatas()[t]->Lxu;
+
+      for (std::size_t idx_row = 0; idx_row < ndxi; idx_row++) {
+        for (std::size_t idx_col = 0; idx_col < ndxi; idx_col++) {
+          // We need the lower triangular matrix
+          if (idx_col > idx_row) {
+            break;
+          }
+          values[idx_value] = obj_factor * datas_[t]->Ldxdx(idx_row, idx_col);
+          idx_value++;
+        }
+      }
+
+      for (std::size_t idx_row = 0; idx_row < nui; idx_row++) {
+        for (std::size_t idx_col = 0; idx_col < ndxi; idx_col++) {
+          values[idx_value] = obj_factor * datas_[t]->Ldxu(idx_col, idx_row);
+          idx_value++;
+        }
+
+        for (std::size_t idx_col = 0; idx_col < nui; idx_col++) {
+          if (idx_col > idx_row) {
+            break;
+          }
+          values[idx_value] = obj_factor * data->Luu(idx_row, idx_col);
+          idx_value++;
+        }
+      }
+      ixu += ndxi + nui;
+    }
+
+    {
+      // Terminal costs
+      const boost::shared_ptr<ActionModelAbstract> &model = problem_->get_terminalModel();
+      const boost::shared_ptr<ActionDataAbstract> &data = problem_->get_terminalData();
+      const std::size_t ndxi = model->get_state()->get_ndx();
+
+      datas_[T_]->dx = Eigen::VectorXd::Map(x + ixu, ndxi);
+      model->get_state()->integrate(xs_[T_], datas_[T_]->dx, datas_[T_]->x);
+      model->calc(data, datas_[T_]->x);
+      model->calcDiff(data, datas_[T_]->x);
```

Comment:
And calcDiff to be revise as well. Most likely Ipopt computes the Jacobians first. So this might be called in one of those functions.

Reply:
Same as previous comments